utilities/roswtf/src/roswtf/network.py

```diff
@@ -49,7 +49,7 @@ def ip_check(ctx):
     local_addrs = rosgraph.network.get_local_addresses()
 
     resolved_ips = [host[4][0] for host in socket.getaddrinfo(socket.gethostname(), 0, 0, 0, socket.SOL_TCP)]
-    global_ips = [ ip for ip in resolved_ips if not ip.startswith('127.') and not ip == '::1']
+    global_ips = [ ip.replace('%', '%%') for ip in resolved_ips if not ip.startswith('127.') and not ip == '::1']
```

Comment:
The substitution problem does not happen in this function. Adding the replace in the list comprehension does not do anything. The fix must happen when the return value of this function is being used with a `%` substitution.